src/Algebra/Graph/Label.hs

```diff
@@ -133,7 +134,7 @@ instance (Num a, Ord a) => Num (NonNegative a) where
         f = fromInteger x
 
     (+) = liftA2 (+)
-    (*) = liftA2 (*)
+    (*) = coerce ((*) :: Extended a -> Extended a -> Extended a)
```

Comment:
Why are `+` and `*` treated differently above?

Reply:
If we can use `(+) = coerce ((+) :: Extended a -> Extended a -> Extended a)` then lets use it to make it more symmetric.